kolibri/plugins/learn/assets/src/state/actions/classesActions.js

```diff
@@ -99,17 +103,18 @@ export function showLessonPlaylist(store, { lessonId }) {
  *
  */
 export function showLessonResourceViewer(store, { lessonId, resourceNumber }) {
-  preparePage(store, {
-    pageName: ClassesPageNames.LESSON_RESOURCE_VIEWER,
-    initialState: {
-      currentLesson: {},
-      // To match expected shape for content-page
-      content: {},
-    },
-  });
+  store.dispatch('CORE_SET_PAGE_LOADING', true);
   return LearnerLessonResource.getModel(lessonId)
     .fetch({}, true)
     ._promise.then(lesson => {
+      preparePage(store, {
```

Comment:
isn't this the only action where moving `preparePage` is absolutely necessary, based on your analysis?

Reply:
It might actually be the only place it isn't. The issue of the state being assigned too early becomes apparent at any point _after_ `content-page` is mounted.